presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTime.java

```diff
@@ -67,6 +83,12 @@ public boolean equals(Object obj)
     @Override
     public String toString()
     {
-        return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(sessionTimeZoneKey.getId())).format(formatter);
+        if (isDeprecatedTimestamp()) {
+            return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(sessionTimeZoneKey.get().getId())).format(formatter);
+        }
+        else {
+            // For testing purposes this should be exact equivalent of TIMESTAMP to VARCHAR cast
```

Comment:
this sounds like a hack, maybe testing infra should have to handle this on its own instead?